gpytorch/models/deep_gps/predictive_deep_gp.py

```diff
@@ -0,0 +1,119 @@
+import torch
+
+from gpytorch import settings
+from gpytorch.distributions import MultitaskMultivariateNormal
+from gpytorch.lazy import BlockDiagLazyTensor
+from gpytorch.likelihoods import GaussianLikelihood, MultitaskGaussianLikelihood
+
+from ..approximate_gp import ApproximateGP
+from .deep_gp import DeepGPLayer as AbstractDeepGPLayer
+
+
+class AbstractPredictiveDeepGPLayer(AbstractDeepGPLayer):
+    def __init__(self, variational_strategy, input_dims, output_dims, num_sample_sites=3, quad_grid=None):
+        super().__init__(variational_strategy, input_dims, output_dims)
+
+        self.num_sample_sites = num_sample_sites
+
+        # Pass in previous_layer.quad_grid if you want to share quad_grid across layers.
+        if quad_grid is not None:
+            self.quad_grid = quad_grid
+        else:
+            self.quad_grid = torch.nn.Parameter(torch.randn(num_sample_sites, input_dims))
+
+    def __call__(self, inputs, are_samples=False, expand_for_quadgrid=True, **kwargs):
+        if isinstance(inputs, MultitaskMultivariateNormal):
+            # inputs is definitely in the second layer, and mean is n x t
+            mus, sigmas = inputs.mean, inputs.variance.sqrt()
+
+            if expand_for_quadgrid:
+                xi_mus = mus.unsqueeze(-3)  # 1 x n x t
```

Comment:
I think we should make this `unsqueeze(0)` for two reasons:

1) It will be more similar to DeepGPs this way (the sample dimension for deep gps is always the left-most batch dimension)
2) It will allow us to do batch DSPPs easier.

Reply:
Yeah this makes sense :+1: